src/UnitConversions.hs

```diff
@@ -54,3 +60,42 @@ convertIngredientToImperial ingr =
     Quart   -> ingr
     Gallon  -> ingr
     Other _ -> ingr
+
+
+convertTemperatureToMetric = convertTemperature C
+convertTemperatureToImperial = convertTemperature F
+
+convertTemperature :: TempUnit -> String -> String
+convertTemperature u s = unpack $ foldl replaceTemperature (pack s) (fmap packText $ fmap convertReplacement (findReplacements s))
+  where packText (s1, s2) = (pack s1, pack s2)
+        replaceTemperature text (old, new) = replace old new text
+        convertReplacement = fmap $ show . toTempUnit u
+
+findReplacements :: String -> [(String, Temperature)]
+findReplacements = map parseRegexResult . findTemperatures
+  where parseRegexResult l = (l!!0, Temperature (read (l!!1)) (parseTempUnit (l!!2)))
+
+findTemperatures :: String -> [[String]]
+findTemperatures s = s =~  "(-?[0-9]{1,3}) ?°?(C|F)"
+
+parseTempUnit :: String -> TempUnit
+parseTempUnit "C" = C
+parseTempUnit "F" = F
+parseTempUnit x = error $ "couldn't parse tempUnit: " ++ x
+
+data Temperature = Temperature Int TempUnit
+instance Show Temperature where
+  show (Temperature value unit) = show value ++ show unit
+data TempUnit = C | F
```

Comment:
It is generally more readable if there is one line of whitespace before and after every declaration (things like `data`, `instance`, and function type signatures `foo :: bar`).

Reply:
There are a few places in this PR where this comment is applicable.

Reply:
Added whitespaces before each function type signature